### PR TITLE
Rework of the lintering and formatting process

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
A second attempt to rebuild the linter and formatter process. I canceled the first one because it did not contain the new flat configuration of ESLint. See also the discussion in PR #115.

**Changes**

- Add stylelint
- Add prettier
- ESLint
  - switch to flat config
  - use recommended rules and define some exceptions (e.g. the use of tabs)
- Add script to fix linter and formatter issues: `npm run lint:fix`
- Handle linter and format issues
- Add `.gitignore` to ignore `node_modules` directory
- Add license name to LICENSE file (like here https://choosealicense.com/licenses/mit/)

There should be no functional changes.

### CODE REVIEW

**Comment by BePo65** [comment]
This looks very good to me.

File: Control.FullScreen.js [hunk 1]
```diff
@@ -142,8 +144,10 @@
 
 		_screenfull: fullscreenAPI,
 
-		onAdd: function (map) {
-			var className = 'leaflet-control-zoom-fullscreen', container, content = '';
+		onAdd(map) {
```
Comment (brunob) [inline Control.FullScreen.js:147]: This change of syntax seems legit as leaflet core doest it on master https://github.com/Leaflet/Leaflet/blob/main/src/control/Control.Attribution.js#L39 but it's not the case in https://github.com/Leaflet/Leaflet/blob/main/src/control/Control.Attribution.js#L39 for instance. Any thoughts about this ?

Comment (KristjanESPERANTO) [reply Control.FullScreen.js:147]: I can't quite follow you. It's the same URL twice 🤔

Comment (brunob) [reply Control.FullScreen.js:147]: wrong paste, my bad ^^

Comment (KristjanESPERANTO) [reply Control.FullScreen.js:147]: Leaflet.heat uses ESLint 1.x, we use 8.x. And the file you linked has not been edited for 6 years. I think we should not use such old code as a reference.

**Review by brunob — APPROVED** [review]

**Comment by brunob** [comment]
Done, thx again @KristjanESPERANTO & @BePo65 :)

**Comment by KristjanESPERANTO** [comment]
> Now we may try to add a CI on the repo in order to run lint on every PR, what do yout think ?

I also think that this would be good step. I don't have an overview of what is currently best practice here. I usually use [husky](https://github.com/typicode/husky), which already prevents me from committing unlinted code. But an action like [this one](https://github.com/marketplace/actions/lint-action) doesn't seem so wrong to me either.

**Comment by brunob** [comment]
> But an action like [this one](https://github.com/marketplace/actions/lint-action) doesn't seem so wrong to me either.

This looks good https://github.com/marketplace/actions/lint-action#javascript-example-eslint-and-prettier and seems almost identical to what we do in https://github.com/leaflet-extras/leaflet-providers/blob/master/.github/workflows/ci.yml

**Comment by brunob** [comment]
Proposition here https://github.com/brunob/leaflet.fullscreen/pull/120

**Comment by brunob** [comment]
> On my repos I use github actions (e.g. [one of my repos](https://github.com/BePo65/mat-tristate-checkbox/blob/main/.github/workflows/node.js.yml)) - that is what you do in pr #120. Why do you add `npm ci`? We don't run or build any code.
> 
> And you can make this workflow mandatory for pul requests (see [github documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches)).

I've tried without npm ci and it fails https://github.com/brunob/leaflet.fullscreen/actions/runs/7614739500/job/20737709233?pr=120 FTR it worked with npm ci https://github.com/brunob/leaflet.fullscreen/actions/runs/7614569335

**Comment by BePo65** [comment]
Sorry, I was wrong. but of course we need `npm ci` otherwise prettier is not installed :-(

**Comment by KristjanESPERANTO** [comment]
> Proposition here https://github.com/brunob/leaflet.fullscreen/pull/120

Good solution :+1:

**Comment by brunob** [comment]
done, i've also publisher a patch release

i think now we can propose to transfer the ownership of this repo to @Leaflet org :)

**Comment by brunob** [comment]
If this plan is not successful, we still have plan B by creating a team of maintainers here :)

**Comment by brunob** [comment]
@KristjanESPERANTO i see that we are still using eslint v8, what do you think about adding something like https://github.com/leaflet-extras/leaflet-providers/blob/master/.github/dependabot.yml in order too keep our dependencies up to date ?

**Comment by KristjanESPERANTO** [comment]
> @KristjanESPERANTO i see that we are still using eslint v8, what do you think about adding something like https://github.com/leaflet-extras/leaflet-providers/blob/master/.github/dependabot.yml in order too keep our dependencies up to date ?

Sure. I've just prepared a PR: #126.

**Pull request merged**